common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -117,76 +119,445 @@ public Collation(
     }
 
     /**
-     * Constructor with comparators that are inherited from the given collator.
+     * collation id (32-bit integer) layout:
+     * bit 31:    0 = predefined collation, 1 = user-defined collation
+     * bit 30-29: 00 = utf8-binary, 01 = ICU, 10 = indeterminate (without spec implementation)
```

Comment:
should we consider using bit 31 for indeterminate collation?

then, indeterminate collation could conveniently be -1 / all 1s in binary

Reply:
My opinion is that it would be better to stick with this naming with following reasoning.

If we used bit 31 for indeterminate collation, we would shrink user collation space because we use additional bit to distinguish between predefined and user collations. It's more convenient to distinguish this at first bit since indeterminate collation falls into predefined space.

Reply:
Schema is outdated.